File: Others/PG_키패드누르기.py
def chk_dist(p1, p2):
    return abs(p1[0] - p2[0]) + abs(p1[1] - p2[1])


def solution(numbers, hand):
    answer = ''
    board = []
    for i in range(4):
        board.append([i * 3 + 1, i * 3 + 2, i * 3 + 3])
    left_finger = [3, 0]
    right_finger = [3, 2]

    idx = 0
    while idx < len(numbers):
        if numbers[idx] % 3 == 0 and numbers[idx] != 0:
            answer += 'R'
            right_finger = [(numbers[idx] // 3) - 1, 2]
        elif numbers[idx] % 3 == 1:
            answer += 'L'
            left_finger = [(numbers[idx] // 3), 0]
        else:
            num_point = []
            if numbers[idx] == 0:
                num_point = [3, 1]
            else:
                num_point = [(numbers[idx] // 3), 1]

            if chk_dist(num_point, left_finger) > chk_dist(num_point, right_finger):
                answer += 'R'
                right_finger = num_point
            elif chk_dist(num_point, left_finger) < chk_dist(num_point, right_finger):
                answer += 'L'
                left_finger = num_point
            else:
                if hand == 'left':
                    answer += 'L'
                    left_finger = num_point
                else:
                    answer += 'R'
                    right_finger = num_point
        idx += 1

    return answer

File: Others/PG_키패드누르기.js
// 프로그래머스 키패드 누르기
// 문제 그대로 풀어나가면 되는 문제
// if문과 for문 만 사용
// map도 써먹어보자...
function solution(numbers, hand) {
  let answer = ''
  let left_num = [1, 4, 7]
  let right_num = [3, 6, 9]
  let left_curr = [3, 0]
  let right_curr = [3, 2]
  for (let num of numbers) {
      if (left_num.includes(num)) {
          answer += 'L'
          left_curr = [parseInt((num-1) / 3), (num-1) % 3]
      } else if (right_num.includes(num)) {
          answer += 'R'
          right_curr =[parseInt((num-1) / 3), (num-1) % 3]
      } else {
          if (num === 0) {
              let left_dist = Math.abs(3 - left_curr[0]) + Math.abs(1 - left_curr[1])
              let right_dist = Math.abs(3 - right_curr[0]) + Math.abs(1 - right_curr[1])
              if (left_dist < right_dist) {
                  answer += 'L'
                  left_curr =[3, 1]
              } else if (left_dist > right_dist) {
                  answer += 'R'
                  right_curr =[3, 1]
              } else {
                  if (hand === "right") {
                      answer += 'R'
                      right_curr =[3, 1]
                  } else {
                      answer += 'L'
                      left_curr =[3, 1]
                  }
              }
              continue
          }
          let left_dist = Math.abs(parseInt((num-1) / 3) - left_curr[0]) + Math.abs((num-1) % 3 - left_curr[1])
          let right_dist = Math.abs(parseInt((num-1) / 3) - right_curr[0]) + Math.abs((num-1) % 3 - right_curr[1])
          if (left_dist < right_dist) {
              answer += 'L'
              left_curr =[parseInt((num-1) / 3), (num-1) % 3]
          } else if (left_dist > right_dist) {
              answer += 'R'
              right_curr =[parseInt((num-1) / 3), (num-1) % 3]                
          } else {
              if (hand === "right") {
                  answer += 'R'
                  right_curr =[parseInt((num-1) / 3), (num-1) % 3]     
              } else {
                  answer += 'L'
                  left_curr =[parseInt((num-1) / 3), (num-1) % 3]
              }
          }
      }
  }
  
  return answer
}